lib/elastic_apm/span.rb

```diff
@@ -61,18 +61,19 @@ def initialize(
 
     # life cycle
 
-    def start(timestamp = Util.micros)
-      @timestamp = timestamp
-
+    def start(start_clock = Util.monotonic_micros)
+      @timestamp = Util.micros
+      @clock = start_clock
```

Comment:
`@clock` is a bit ambiguous of a name, as it's actually the starting micros. I don't necessarily have any better ideas though.
Maybe `@start_micros`?

Reply:
It's referring to the _clock_ time so I think it makes sense to have clock in there. It might be because I'm differentiating it from what used to be but just _micros_ tells me it's related to ordinary `Time`.

Reply:
Could call both the ivar and the argument `clock_start` instead.